pandas/util/_decorators.py

```diff
@@ -245,13 +245,13 @@ def wrapper(*args, **kwargs) -> Callable[..., Any]:
     return decorate
 
 
-def doc(*args: Union[str, Callable], **kwargs: str) -> Callable[[F], F]:
+def doc(*args: Union[str, Callable], **kwargs: str) -> Callable[[Callable], Callable]:
```

Comment:
For the names of the parameters, I think a descriptive name is better in this case. I think `args` and `kwargs` are great names when they are generic, you can pass whatever, and they are passed to another function mostly.

But in this case, they have very specific usage, and they are very different between them.

I think something like `*docs` or `*docs_or_objs` and `**params` could be ok. May be you think of something better.

Reply:
> But in this case, they have very specific usage, and they are very different between them.
I like this point! Agree that we should be specific in this case.

For the names, I think `docs` and `params` are good. I also want to add some other candidates here:
1. use `supplements` to replace `args`
2. use `fillers` to replace `kwargs`

I am not a native English speaker, so I am not sure if those words also make sense to you and other developers. Please let me know which one you like and I will be more than happy to make the change.